data/tasks.js

```diff
@@ -0,0 +1,192 @@
+import { tasksCollection } from "../config/mongoCollections.js";
+import { ObjectId } from "mongodb";
+import utils from "../utils/utils.js";
+import { usersCollection } from "../config/mongoCollections.js";
+
+const tasksDataFunctions = {
+  async getTaskById(id) {
+    utils.checkObjectIdString(id);
+    id = id.trim();
+    const tasks = await tasksCollection();
+    const task = await tasks.findOne({ _id: new ObjectId(id) });
+
+    if (!task) {
+      throw new Error(`Task with ID ${id} not found`);
+    }
+
+    return task;
+  },
+  /**
+   * @param {string} userId - The user ID to which the task is associated
+   * @param {string} title - The title of the task
+   * @param {string} textBody - The body text of the task
+   * @param {Date} dateAddedTo - The date when the task was added to the collection
+   * @param {Date} dateDueOn - The date when the task is due
+   * @param {number} priority - The priority of the task (1, 2, or 3)
+   * @param {string} tag - The custom tag for the task
+   */
+  async createTask(
+    userId,
```

Comment:
Please trim the user id, " string " the space would cause issue in converting to object id. Make sure any string you have you trim it in you date model.

Reply:
trimmed in route